kernel/src/snapshot.rs

```diff
@@ -71,6 +72,26 @@ impl Snapshot {
         Self::try_new_from_log_segment(table_root, log_segment, engine)
     }
 
+    /// Create a new [`Snapshot`] instance from an existing [`Snapshot`]. This is useful when you
+    /// already have a [`Snapshot`] lying around and want to do the minimal work to 'update' the
+    /// snapshot to a later version.
+    ///
+    /// # Parameters
+    ///
+    /// - `existing_snapshot`: reference to an existing [`Snapshot`]
+    /// - `engine`: Implementation of [`Engine`] apis.
+    /// - `version`: target version of the [`Snapshot`]. None will create a snapshot at the latest
+    ///   version of the table.
+    pub fn new_from(
+        existing_snapshot: &Snapshot,
+        engine: &dyn Engine,
+        version: Option<Version>,
+    ) -> DeltaResult<Self> {
```

Comment:
Seems like the method should take+return `Arc<Snapshot>` so we have the option to return the same snapshot if we determine it is still fresh?

Reply:
Maybe even do
```rust
pub fn refresh(self: &Arc<Self>, ...) -> DeltaResult<Arc<Self>>
```
(this would have slightly different intuition than `new_from` -- `refresh` specifically assumes I want a newer snapshot, if available, and attempting to request an older `version` may not even be legal; I'm not sure if it would even make sense to pass an upper bound version for a refresh operation)